pkg/storage/ingest/reader_test.go

```diff
@@ -155,40 +155,35 @@ func TestPartitionReader_ConsumerStopping(t *testing.T) {
 
 	_, clusterAddr := testkafka.CreateCluster(t, partitionID+1, topicName)
 
-	invocations := atomic.NewInt64(0)
-	blockingConsumer := newTestConsumer(0)
-	consumer := consumerFunc(func(ctx context.Context, records []record) error {
-		invocations.Inc()
-		return blockingConsumer.consume(ctx, records)
+	// consumerErrs will store the last error returned by the consumer; its initial value doesn't matter, but it must be non-nil.
+	consumerErrs := atomic.NewError(errors.New("dummy error"))
+	consumeCalls := make(chan chan error)
+	consumer := consumerFunc(func(context.Context, []record) (err error) {
+		defer consumerErrs.Store(err)
+
+		call := make(chan error, 1)
+		consumeCalls <- call
+		return <-call
```

Comment:
can we just share a `chan error` instead of a `chan chan error`? From what I can see `consumeCalls` is only read from and written to once